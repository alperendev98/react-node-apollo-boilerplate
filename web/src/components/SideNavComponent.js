import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';

import logoIcon from '../assets/images/logo.svg';
import '../styles/base/base.css';
import '../styles/components/sidenavComponent.css';

const BaseClasses = getBEMClasses(['base']);
const SideNavClasses = getBEMClasses(['side-nav']);

class SideNavComponent extends React.Component {
	navigateTo(route) {
		const { history } = this.props;

		history.push(route);
	}

	render() {
		const { showSideNav, hideSideNav } = this.props;

		// TO DO: test mouseenter and mouseleave events
		return (
			<div
				onMouseEnter={showSideNav}
				onMouseLeave={hideSideNav}
				className={SideNavClasses('container')}
			>
				<div className={SideNavClasses('title')}>Manage</div>
				<div className={SideNavClasses('item')}>Database</div>
				{/* TODO: write tests for onClick */}
				<div
					onClick={() => this.navigateTo('/admin/equipment')}
					className={SideNavClasses('sub-item', 'equipment')}
				>
					Equipment
				</div>
				<div
					onClick={() => this.navigateTo('/admin/utilities')}
					className={SideNavClasses('sub-item', 'utilities')}
				>
					Utilities
				</div>
				<div className={SideNavClasses('item')}>Proposals</div>
				<div className={SideNavClasses('sub-item')}>Contingencies</div>
				<div className={SideNavClasses('sub-item')}>Cost Per Watt</div>
				<div className={SideNavClasses('sub-item')}>Sales Tax</div>
				<div className={SideNavClasses('sub-item')}>Tax Credits</div>
				<div className={SideNavClasses('sub-item')}>Tax Bracket</div>
				<div className={SideNavClasses('sub-item')}>Depreciation Schedules</div>
				<div className={SideNavClasses('sub-item')}>Production Offsets</div>
				<div className={SideNavClasses('sub-item')}>Irradiance</div>
				<div className={SideNavClasses('sub-item')}>Templates</div>
				<div className={SideNavClasses('item')}>Finance</div>
				<div className={SideNavClasses('sub-item')}>PPA's</div>
				<div className={SideNavClasses('sub-item')}>Financing</div>
				<div className={SideNavClasses('item')}>Post Sale</div>
				<div className={SideNavClasses('sub-item')}>Project Stage</div>
			</div>
		);
	}
}

export default SideNavComponent;
