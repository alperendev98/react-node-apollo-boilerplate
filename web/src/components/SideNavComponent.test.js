import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import SideNavComponent from './SideNavComponent';

jest.mock('./SideNavComponent', () => {
	return () => <div className="side-nav__container" />;
});

const mockStore = configureStore();

describe('SideNavComponent', () => {
	const initialStore = {};
	const initialProps = {
		history: {
			push: jest.fn()
		},
		showSideNav: jest.fn(),
		hideSideNav: jest.fn()
	};
	let wrapper, component;

	beforeEach(() => {
		const store = mockStore(initialStore);
		wrapper = shallow(<SideNavComponent />);
		component = mount(<SideNavComponent {...initialProps} store={store} />);
	});

	it('renders SideNavComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should render SideNavComponent with .side-nav__container class', () => {
		expect(component.find('.side-nav__container').length).toBe(1);
	});

	it('should match previos snapshot for SideNavComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
