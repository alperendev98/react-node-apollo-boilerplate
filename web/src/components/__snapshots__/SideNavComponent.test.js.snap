// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavComponent should match previos snapshot for SideNavComponent 1`] = `
<div
  className="side-nav__container"
/>
`;
